connectivity/FEATURE_BLE/source/cordio/source/BLEInstanceBaseImpl.cpp

```diff
@@ -17,9 +17,26 @@
  */
 
 #include "mbed.h"
-#include "us_ticker_api.h"
+#include "platform/CriticalSectionLock.h"
+#include "hal/us_ticker_api.h"
+#include "platform/mbed_assert.h"
+
 #include "ble/BLE.h"
-#include "CriticalSectionLock.h"
+#include "ble/driver/CordioHCIDriver.h"
+
+#include "source/pal/PalAttClient.h"
```

Comment:
why are headers in the source folder?

Reply:
Because there is private headers and public headers. public headers goes into `include/<module name>` while private headers goes into `source`

Reply:
that's not how we agreed to do it, private headers go into internal, did you ask core about this?

Reply:
Yes I did ask Core about this. Obviously one of the precondition for this is that private headers are **not** included by public ones which wasn't possible with the previous separation.